tools/src/main/java/org/apache/kafka/tools/DelegationTokenCommand.java

```diff
@@ -0,0 +1,306 @@
+import org.apache.kafka.clients.admin.CreateDelegationTokenResult;
+import org.apache.kafka.clients.admin.DescribeDelegationTokenOptions;
+import org.apache.kafka.clients.admin.DescribeDelegationTokenResult;
+import org.apache.kafka.clients.admin.ExpireDelegationTokenOptions;
+import org.apache.kafka.clients.admin.ExpireDelegationTokenResult;
+import org.apache.kafka.clients.admin.RenewDelegationTokenOptions;
+import org.apache.kafka.clients.admin.RenewDelegationTokenResult;
+import org.apache.kafka.common.security.auth.KafkaPrincipal;
+import org.apache.kafka.common.security.token.delegation.DelegationToken;
+import org.apache.kafka.common.security.token.delegation.TokenInformation;
+import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.common.utils.SecurityUtils;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.server.util.CommandDefaultOptions;
+import org.apache.kafka.server.util.CommandLineUtils;
+
+public class DelegationTokenCommand {
+    public static void main(String... args) {
+        Exit.exit(mainNoExit(args));
+    }
+
+    static int mainNoExit(String... args) {
+        try {
+            execute(args);
+            return 0;
+        } catch (TerseException e) {
+            System.err.println(e.getMessage());
+            return 1;
+        } catch (Throwable e) {
+            System.err.println("Error while executing delegation token command : " + e.getMessage());
+            System.err.println(Utils.stackTrace(e));
+            return 1;
+        }
+    }
+
+    static void execute(String... args) throws Exception {
+        Admin adminClient = null;
+        try {
+            DelegationTokenCommandOptions opts = new DelegationTokenCommandOptions(args);
+            CommandLineUtils.maybePrintHelpOrVersion(opts, "This tool helps to create, renew, expire, or describe delegation tokens.");
+
+            // should have exactly one action
+            int numberOfAction = 0;
+            for (Boolean opt : new Boolean[]{opts.hasCreateOpt(), opts.hasRenewOpt(), opts.hasExpireOpt(), opts.hasDescribeOpt()}) {
+                if (opt) {
+                    numberOfAction++;
+                }
+            }
+            if (numberOfAction != 1) {
+                CommandLineUtils.printUsageAndExit(opts.parser, "Command must include exactly one action: --create, --renew, --expire or --describe");
+            }
+
+            opts.checkArgs();
+
+            adminClient = createAdminClient(opts);
+
+            if (opts.hasCreateOpt()) {
+                createToken(adminClient, opts);
+            } else if (opts.hasRenewOpt()) {
+                renewToken(adminClient, opts);
+            } else if (opts.hasExpireOpt()) {
+                expireToken(adminClient, opts);
+            } else if (opts.hasDescribeOpt()) {
+                describeToken(adminClient, opts);
+            }
+
+        } finally {
+            if (adminClient != null)
+                adminClient.close();
```

Comment:
Since Admin implements AutoCloseable could you use https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html?